[WIP] : add support for github actions

indiVar0508 opened the pull request.

indiVar0508 commented:
Hi @ananyo2012 , 

Added coverage step and to report it in PR comment which will be update for each run.

I tried adding testing-report but it seems to be failing ig due to the format which is not acceptable by the [test-reporter](https://github.com/marketplace/actions/test-reporter) action, getting error in reading file with jest-junit.
ref: https://github.com/indiVar0508/junction/actions/runs/5365317262/jobs/9734065641
.
A workaround ig is to use java-junit but requires to [unittest-xml-reporting](https://pypi.org/project/unittest-xml-reporting/) to generate report which still has some limitations 
ref: https://github.com/dorny/test-reporter/issues/244

ananyo2012 commented:
@indiVar0508 Please check https://github.com/marketplace/actions/publish-test-results as an alternative for test-reporting

indiVar0508 commented:
Thanks @ananyo2012 ,
I have added same in this PR you can check once ig some permissions from project is required as i saw last time it failed write comment however it was able to write comment in my forked project.
Do let know if anything else is needed to be added or some other issue that i can take up

ananyo2012 commented:
@indiVar0508 You need to create a separate workflow for Publishing Results - https://github.com/EnricoMi/publish-unit-test-result-action/blob/v2.8.0/README.md#support-fork-repositories-and-dependabot-branches . Let's Name the first Workflow as 'CI' and second one as 'Test Report'.

Also the Pull request comments should only run during Pull request events, not for master branch.

indiVar0508 commented:
Added as per comments @ananyo2012 
- renamed the respective workflows
- encorporated steps metioned in the links
- comment for coverage reports will posted for pull request event and publish test result seems to already handle this
